### PR TITLE
Support booleans in routing path

### PR DESCRIPTION
**Pull request by felixbarny**
This doesn't add full support for booleans as dimensions (#111338). But it does make it possible to send boolean values that are within the routing_path. These booleans need to be converted to a keyword field in the mappings.

### CODE REVIEW

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml [hunk 1]
```diff
@@ -1052,3 +1052,87 @@ dimensions with ignore_malformed and ignore_above:
   - length: { aggregations.keyword_dims.buckets: 1 }
   - match: { aggregations.keyword_dims.buckets.0.key: "foo" }
   - match: { aggregations.keyword_dims.buckets.0.doc_count: 2 }
+
+---
+non string dimension fields:
+  - requires:
+      cluster_features: ["mapper.pass_through_priority"]
```
Comment (felixbarny) [inline modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml:1059]: Do I need to create a new cluster feature for this? If so, should that be in a new class `org.elasticsearch.cluster.routing.RoutingFeatures`?

**Review by kkrik-es — COMMENTED** [review]
Looks good, let's also add a unit test like `testRoutingPathNumbersInSource`.

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml (hunk 1 above)
Comment (kkrik-es) [reply modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml:1059]: Yeah, it's fairly simple. See https://github.com/elastic/elasticsearch/pull/107567/files

Comment (felixbarny) [reply modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml]: done

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml [hunk 2]
```diff
@@ -1052,3 +1052,87 @@ dimensions with ignore_malformed and ignore_above:
...
+          - '{ "create": { "dynamic_templates": { "metrics.data": "counter_metric" } } }'
+          - '{ "@timestamp": "2023-09-01T13:03:08.138Z", "metrics.data": "10", "attributes.string": "foo", "attributes.boolean": true, "attributes.integer": 1, "attributes.float": 1.1, "attributes.host.ip": "127.0.0.1" }'
+  - is_false: errors
+
+  - do:
+      search:
+        index: k9s
+        body:
+          size: 1
+  - match: { hits.total.value: 1 }
```
Comment (kkrik-es) [inline modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml:1123]: Nit: let's also check that the fields are included in the response as expected, and maybe add filtering on the non-string dimensions.

**Review by kkrik-es — APPROVED** [review]
Thanks Felix!

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml (hunk 2 above)
Comment (felixbarny) [reply modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml:1142]: Good idea. Implemented in a2636f5.

**Pull request merged**